### PR TITLE
adding retry

### PR DESCRIPTION
**Pull request by calellowitz**
@biyeun
This makes async builds (those triggered by updating the phone to the latest saved version of an app), retry if an older build is ongoing. I picked some relatively arbitrary numbers for retries and delay, but wanted to make sure the total time spent retrying was as long as the timeout for the lock.

cc: @amstone326 FYI

### CODE REVIEW

File: corehq/apps/app_manager/tasks.py [hunk 1]
```diff
@@ -12,7 +12,7 @@ def create_user_cases(domain_name):
         sync_usercase(user)
 
 
-@serial_task('{app._id}', max_retries=0, timeout=60*60)
+@serial_task('{app._id}', max_retries=30, default_retry_delay=120, timeout=60*60)
```
Comment (dimagimon) [inline corehq/apps/app_manager/tasks.py]: E226 missing whitespace around arithmetic operator

**Comment by calellowitz** [comment]
switched key to include version because it should be safe and will stop the waiting on a lock for retries. including @gcapalbo and @snopoke since they were on the first pr and may have context

File: corehq/apps/app_manager/tasks.py [hunk 2]
```diff
@@ -12,7 +12,7 @@ def create_user_cases(domain_name):
         sync_usercase(user)
 
 
-@serial_task('{app._id}', max_retries=0, timeout=60*60)
+@serial_task('{app._id}-{app.version}', max_retries=0, timeout=60*60)
```
Comment (dimagimon) [inline corehq/apps/app_manager/tasks.py]: E226 missing whitespace around arithmetic operator

**Review by snopoke — APPROVED** [review]

**Comment by gcapalbo** [comment]
@calellowitz can you explain where the retries are happening? it looks like this change just allows concurrent builds of different versions of the same app and celery still won't retry anything (maybe that's what's intended, just want to make sure).

**Review by biyeun — APPROVED** [review]

**Comment by calellowitz** [comment]
@gcapalbo Im sorry, I changed the approach in the second commit and did not explain that well. The initial bug was that async builds would be missed, if subsequent builds were kicked off while one was completing due to the lock on app id. Adding version to that key should fix the problem, since we only want one build per version anyway, and now we dont need to worry about waiting for retries.

**Comment by gcapalbo** [comment]
:+1: sounds good, thanks for clarifying

**Pull request merged**